src/simulation/decision_model_simulation_results.jl

```diff
@@ -336,12 +269,24 @@ function _read_results(
     end
     existing_keys = list_result_keys(res, first(result_keys))
     _validate_keys(existing_keys, result_keys)
-    cached_results = get_cached_results(res, first(result_keys))
+    cached_results = get_cached_results(res, eltype(result_keys))
     if _are_results_cached(res, result_keys, timestamps, keys(cached_results))
-        @debug "reading results from SimulationsResults cache"
+        @debug "reading results from SimulationsResults cache"  # NOTE tests match on this
```

Comment:
remove `# Note`

Reply:
As a developer I find this useful. It calls my attention to it.

Reply:
ok, lets leave it